### PR TITLE
Make models.wave compatible with all array contexts

### PR DESCRIPTION
**Pull request by kaushikcfd**

### CODE REVIEW

**Review by thomasgibson — APPROVED** [review]
Thanks!

File: grudge/shortcuts.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 """
 
 
-def set_up_rk4(field_var_name, dt, fields, rhs, t_start=0):
+def set_up_rk4(field_var_name, dt, fields, rhs, t_start=0.0):
```
Comment (inducer) [inline grudge/shortcuts.py:26]: @kaushikcfd Out of curiosity: How come this mattered?

Comment (kaushikcfd) [reply grudge/shortcuts.py:26]: `t` is one of the arguments to the `rhs`. Passing an `int64` type at first and then `float64` leads to compiling 2 expression graphs. This was adding up to the setup times in my experiments and hence got rid of it.

Comment (inducer) [reply grudge/shortcuts.py:26]: Ah. Fair enough, thanks for explaining!

**Pull request merged**